src/ocean/meta/traits/Basic.d

```diff
@@ -92,6 +92,46 @@ unittest
     static assert (!isCharType!(byte));
 }
 
+/*******************************************************************************
+
+    Checks if a given type is a string type or not.
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is a string type, false otherwise.
+
+*******************************************************************************/
+
+public template isStringType( T )
+{
+    static immutable bool isStringType =
+        is( T : char[] )
+        || is( T : dchar[] )
+        || is( T : wchar[] )
+        || is( T : istring )
+        || is( T : cstring )
+        || is( T : mstring );
```

Comment:
Weeeell there's an interesting question here. Is a `Typedef` (or anything else that implicitly converts to a string) a string type ? If it is, you can just go with:
```
is(T : const(char)[]) || is(T : const(wchar)[]) || is(T : const(dchar)[])
```
Note however that we removed non-UTF8 support from many places in Ocean so you might just be fine reducing it to one condition.

But if `Typedef` should not match `isStringType`, then you need a long list of `is(T == ...)`.

Reply:
Looking at other templates that were put in `ocean.meta.traits.Basic` from `ocean.core.Traits`, it seems that the functionality of a given template in `ocean.meta.traits.Basic` should be the same as that in `ocean.core.Traits`. So, I do not think the implementation of the template here should try to consider `Typedef`s (or other implicitly convertible types) as `string`s. What do you suggest?

Reply:
It could create problems for people porting from the deprecated code if the new code behaves differently I guess.

Reply:
>  it seems that the functionality of a given template in ocean.meta.traits.Basic should be the same as that in ocean.core.Traits. So, I do not think the implementation of the template here should try to consider Typedefs (or other implicitly convertible types) as strings. What do you suggest?

Well... The implementation you copied does support `Typedef`. Try `isStringType` on a `Typedef!string`. The question was basically: Should it be `is(T == v)` or `is(T : v)`.

Reply:
[As Mihails mentioned](https://github.com/sociomantic-tsunami/ocean/pull/713#issuecomment-498819829), I'll update the deprecation message.